extension/src/saturation.rs

```diff
@@ -0,0 +1,114 @@
+use pgx::*;
+
+/// Saturating Math for Integers
+
+/// Computes x+y, saturating at the numeric bounds instead of overflowing
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+fn addition_saturate_to_min(x: i32, y: i32) -> i32 {
+    x.saturating_add(y)
+}
+
+/// Computes x+y, saturating at 0 for the minimum bound instead of i32::MIN
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+fn addition_saturate_to_zero(x: i32, y: i32) -> i32 {
```

Comment:
Maybe `saturating_add_pos` or `pos_saturating_add`?

Reply:
+1 for `saturating_add_pos`